engine/class_modules/priest/sc_priest_holy.cpp

```diff
@@ -12,130 +12,359 @@ namespace priestspace
+  double cost() const override
   {
-    parse_options( options_str );
+    if ( priest().buffs.apotheosis->check() )
+    {
+      return 0;
+    }
 
-    harmful = false;
+    return priest_heal_t::cost();
   }
 
   void execute() override
   {
-    priest_spell_t::execute();
-
-    priest().buffs.apotheosis->trigger();
-    sim->print_debug( "{} starting Apotheosis. ", priest() );
+    priest_heal_t::execute();
+    if ( priest().talents.holy.divine_image.enabled() )
+    {
+      priest().buffs.divine_image->trigger();
+    }
   }
 };
 
-struct holy_fire_t final : public holy_fire_base_t
+struct holy_word_serenity_t final : public priest_heal_t
 {
-  timespan_t manipulation_cdr;
-
-  holy_fire_t( priest_t& player, util::string_view options_str )
-    : holy_fire_base_t( "holy_fire", player, player.find_class_spell( "Holy Fire" ) ),
-      manipulation_cdr( timespan_t::from_seconds( priest().talents.manipulation->effectN( 1 ).base_value() / 2 ) )
-
+  holy_word_serenity_t( priest_t& p, util::string_view options_str )
+    : priest_heal_t( "holy_word_serenity", p, p.talents.holy.holy_word_serenity )
   {
     parse_options( options_str );
+    harmful = false;
+  }
 
-    auto rank2 = priest().find_specialization_spell( "Holy Fire", "Rank 2" );
-    if ( rank2->ok() )
+  double cost() const override
+  {
+    if ( priest().buffs.apotheosis->check() )
     {
-      dot_max_stack += as<int>( rank2->effectN( 2 ).base_value() );
+      return 0;
     }
+
+    return priest_heal_t::cost();
   }
 
   void execute() override
   {
-    priest_spell_t::execute();
-
-    if ( priest().talents.manipulation.enabled() )
+    priest_heal_t::execute();
+    if ( priest().talents.holy.divine_image.enabled() )
     {
-      priest().cooldowns.mindgames->adjust( -manipulation_cdr );
+      priest().buffs.divine_image->trigger();
     }
   }
 };
 
-struct holy_word_chastise_t final : public priest_spell_t
+struct apotheosis_t final : public priest_spell_t
 {
-  holy_word_chastise_t( priest_t& player, util::string_view options_str )
-    : priest_spell_t( "holy_word_chastise", player, player.find_class_spell( "Holy Word: Chastise" ) )
+  apotheosis_t( priest_t& p, util::string_view options_str )
+    : priest_spell_t( "apotheosis", p, p.talents.holy.apotheosis )
   {
     parse_options( options_str );
+    harmful = false;
   }
-  double cost() const override
+
+  void execute() override
   {
-    if ( priest().buffs.apotheosis->check() )
-      return 0;
-    return priest_spell_t::cost();
+    priest_spell_t::execute();
+
+    priest().buffs.apotheosis->trigger();
+    sim->print_debug( "starting apotheosis. " );
+    priest().cooldowns.holy_word_chastise->reset( false );
+    priest().cooldowns.holy_word_serenity->reset( false, -1 );
+    priest().cooldowns.holy_word_sanctify->reset( false, -1 );
+    sim->print_debug( "apotheosis reset holy word cooldowns." );
```

Comment:
any reason in particular for these debug messages?

Reply:
Primarily for my testing, removed.